### DISCUSSION TITLE
Securing self hosted version of draw.io

### DISCUSSION THREAD

**Discussion by schaithanya** [jgraph/drawio#3204] [General]
Hello team,

Thanks for the good work on the tool , I just wanted some design ideas on the below problem

Problem statement:

I have deployed the docker image of draw.io into azure app service.
This self hosted version of draw.io editor is used in our application by embedding it in an iframe.

Currently the self hosted version is publicly available, we are planning to add some custom libraries containing application specific data. 

Any ideas how to limit the public access and also allow embedding of the tool?

**Comment by davidjgraph** [comment]
This isn't really specific to draw.io, this is a more generic server/network question.

**Comment by Atmaks** [comment]
@schaithanya were you able to find a solution? If so, could you please describe it in general terms? Facing the same issue right now.
